src/decimal.rs

```diff
@@ -0,0 +1,239 @@
+        Ok(self.value.cmp(&rhs))
+    }
+
+    fn to_bool(&self) -> bool {
+        !self.value.is_zero()
+    }
+
+    fn plus(&self, heap: &'v Heap) -> starlark::Result<Value<'v>> {
+        Ok(alloc_decimal(heap, self.value))
+    }
+
+    fn minus(&self, heap: &'v Heap) -> starlark::Result<Value<'v>> {
+        Ok(alloc_decimal(heap, -self.value))
+    }
+
+    fn add(&self, rhs: Value<'v>, heap: &'v Heap) -> Option<starlark::Result<Value<'v>>> {
+        Some(decimal_from_value(rhs, "+").map(|rhs| {
+            alloc_decimal(heap, self.value + rhs)
+        }))
+    }
+
+    fn radd(&self, lhs: Value<'v>, heap: &'v Heap) -> Option<starlark::Result<Value<'v>>> {
+        Some(decimal_from_value(lhs, "+").map(|lhs| {
+            alloc_decimal(heap, lhs + self.value)
+        }))
+    }
+
+    fn sub(&self, rhs: Value<'v>, heap: &'v Heap) -> starlark::Result<Value<'v>> {
+        let rhs = decimal_from_value(rhs, "-")?;
+        Ok(alloc_decimal(heap, self.value - rhs))
+    }
+
+    fn mul(&self, rhs: Value<'v>, heap: &'v Heap) -> Option<starlark::Result<Value<'v>>> {
+        Some(decimal_from_value(rhs, "*").map(|rhs| {
+            alloc_decimal(heap, self.value * rhs)
+        }))
+    }
+
+    fn rmul(&self, lhs: Value<'v>, heap: &'v Heap) -> Option<starlark::Result<Value<'v>>> {
+        Some(decimal_from_value(lhs, "*").map(|lhs| {
+            alloc_decimal(heap, lhs * self.value)
+        }))
+    }
+
+    fn div(&self, rhs: Value<'v>, heap: &'v Heap) -> starlark::Result<Value<'v>> {
+        let rhs = decimal_from_value(rhs, "/")?;
+        if rhs.is_zero() {
+            return Err(ValueError::DivisionByZero.into());
+        }
+        Ok(alloc_decimal(heap, self.value / rhs))
+    }
+
+    fn floor_div(&self, rhs: Value<'v>, heap: &'v Heap) -> starlark::Result<Value<'v>> {
+        let rhs = decimal_from_value(rhs, "//")?;
+        if rhs.is_zero() {
+            return Err(ValueError::DivisionByZero.into());
+        }
+        let division = self.value / rhs;
+        Ok(alloc_decimal(heap, division.floor()))
+    }
+
+    fn percent(&self, rhs: Value<'v>, heap: &'v Heap) -> starlark::Result<Value<'v>> {
+        let rhs = decimal_from_value(rhs, "%")?;
+        if rhs.is_zero() {
+            return Err(ValueError::DivisionByZero.into());
+        }
+        Ok(alloc_decimal(heap, self.value % rhs))
+    }
+
+    fn write_hash(&self, hasher: &mut StarlarkHasher) -> starlark::Result<()> {
+        self.value.hash(hasher);
+        Ok(())
+    }
+}
+
+/// Convert DecimalValue to Python Decimal object
+pub fn decimal_to_python(decimal_value: &DecimalValue) -> PyResult<Py<PyAny>> {
+    let decimal_str = decimal_value.value.to_string();
+    Python::attach(|py| {
+        let decimal_module = py.import("decimal")?;
+        let decimal_type = decimal_module.getattr("Decimal")?;
+        let py_decimal = decimal_type.call1((decimal_str,))?;
+        Ok(py_decimal.into())
+    })
```

Comment:
This should probably try to match the precision set on the Rust side, otherwise any operations on the Python side will have surprising rounding behavior.